src/platform.ts

```diff
@@ -130,13 +130,19 @@ export class HomebridgeMagichomeDynamicPlatform implements DynamicPlatformPlugin
           newDevices++;
           
         } else {
-          if(existingAccessory.context.device.deviceData == undefined) {
-            this.log.warn('Warning! Outdated object data detected. Attempting to repair');  
+
+          // Trigger for an object repair is missing 'controllerLogicType' property
+          if(!existingAccessory.context.device?.lightParameters?.controllerLogicType) {
+            const { uniqueId = 'n/a' } = existingAccessory.context.device || {};
+            this.log.warn(`Warning! Outdated object data detected. Attempting to repair "${uniqueId}"`);  
             const initialState = await this.getInitialState (deviceDiscovered.ipAddress, 10000);
             if( initialState == undefined){
               return undefined;
             }
-            const oldName = existingAccessory.context.displayName;
+
+            const oldName = existingAccessory.context.displayName || 
```

Comment:
in my setup, some devices did not have a displayName, so I added a backup name and a fallback.